elasticsearch/src/main/java/org/hibernate/search/elasticsearch/util/impl/GsonHttpEntity.java

```diff
@@ -199,17 +208,18 @@ private void attemptOnePassEncoding() {
 		// as it's not set yet.
 		try {
 			triggerFullWrite();
+			if ( nextBodyToEncodeIndex == bodyParts.size() ) {
+				writer.flush();
+				// The buffer's current content size is the final content size,
+				// as we know the entire content has been encoded already,
+				// and we also know no content was consumed from the buffer yet.
+				hintContentLength( writer.byteBufferContentSize() );
+			}
 		}
 		catch (IOException e) {
 			// Unlikely: there's no output buffer yet!
```

Comment:
I'll amend this comment, as we know now that it might be caused by encoding problems too.